code-studio/src/js/components/progress/stage_details.jsx

```diff
@@ -0,0 +1,38 @@
+/* global React, dashboard */
+
+var STAGE_TYPE = require('./types').STAGE_TYPE;
+
+/**
+ * Stage progress component used in level header and course overview.
+ */
+var CourseProgressRow = React.createClass({
+  propTypes: {
+    stage: STAGE_TYPE
+  },
+
+  render: function () {
+    var items = this.props.stage.levels.map(function (level) {
+      return (
+        <div key={level.id}>
+          <a href={level.url}>
+            <div className={`level-${level.id} level_link ${level.status || 'not_tried'}`}>
```

Comment:
Waiting on this for now since there are still questions about the new progress JSON format.